templates/common/baremetal/files/NetworkManager-clean-initrd-state.yaml

```diff
@@ -0,0 +1,75 @@
+mode: 0755
+path: "/usr/local/bin/nm-clean-initrd-state.sh"
+contents:
+  inline: |
+    #!/bin/bash
+
+    NM_DEVICES_DIR=/run/NetworkManager/devices
+    NM_CONN_DIR=/run/NetworkManager/system-connections
+
+    [ ! -e "$NM_DEVICES_DIR" ] || [ ! -e "$NM_CONN_DIR" ] && exit
+
+    # Some deployments boot with ip=(dhcp|dhcp6) kernel argument. From this, a
+    # NM dracut module generates a default connection profile. If rd.neednet=1
+    # is in the kernel command line or another dracut module needs the network,
+    # NM is also run by dracut, activating the previously generated profile.
+    # This activation generates NM state which makes that profile to be
+    # re-activated by the NM service. And this has the effect that other
+    # profiles deployed by the user for the same set of devices are ignored.
+
+    # The intention here is to remove the state that was generated with the
+    # activation of those profiles during dracut execution. Then when NM service
+    # runs, the profiles generated by dracut, along with other profiles
+    # deployed, are evaluated towards finding the most appropriate profile to
+    # connect a device with.
+
+    # check that we are doing this for the expected kernel arguments
+    CMD_LINE_REGEX="(^|\s+)ip=(dhcp|dhcp6|dhcp,dhcp6|dhcp6,dhcp)(\s+|$)"
+    grep -Eqs "$CMD_LINE_REGEX" /proc/cmdline || exit
+
+    for device in "${NM_DEVICES_DIR}"/*; do
+      [ -e "$device" ] || continue
+
+      # get the uuid of the profile the device has been activated with
+      device_profile_uuid=$(sed -nr '/^\[device\]/,/^\[/{/^connection-uuid\s*=/{s/[^=]+\s*=\s*//;P}}' "$device")
+
+      # the device was not activated with any profile, ignore
+      [ -n "$device_profile_uuid" ] || continue
+
+      # find the profile by uuid
+      for profile in "${NM_CONN_DIR}"/*; do
+        profile_uuid=$(sed -nr '/^\[connection\]/,/^\[/{/^uuid\s*=/{s/[^=]+\s*=\s*//;P}}' "$profile")
+        [ "$profile_uuid" = "$device_profile_uuid" ] && break
+      done
+
+      # profile not found, ignore
+      [ "$profile_uuid" = "$device_profile_uuid" ] || continue
+
+      # profile not generated by nm-initrd-generator, ignore
+      # only check if the key is set (from NM 1.32.4)
```

Comment:
Don't we need this to work in 1.30?

Reply:
Yes. So for that version the key is not there and we don't do that extra precautionary check. It does not prevent it from working.